### PR TITLE
fix(BetterSliding): 修复库存为零时 max_quantity=0 被max_quantity < 1硬拦截导致任务失败 (#3108)

### PR DESCRIPTION
**Pull request by ztg-pty**
修复了#3108 提到的零库存情况直接结束任务的问题，具体修改如下：
安全网： resolveMaxQuantityNext()：maxQuantity==0 在最前面直接返回 Done
日志区分"零库存"和"超限"，这样方便排查
回归测试：handlers_test.go：6 条回归用例覆盖零库存+边界值
(应该还可以在完善的，先看看效果)

## 由 Sourcery 提供的总结

在 BetterSliding 中处理零库存场景时，不再视为错误，并确保数据流可以干净地结束。

错误修复：
- 在 `resolveMaxQuantityNext` 中将 `maxQuantity == 0` 视为立即进入 `Done` 状态，以防止在库存为零时任务失败或崩溃。
- 在 `handleGetMaxQuantity` 中将零库存条件标记为已超出（exceeded），使流程通过 override（覆盖）路径退出，而不是错误地继续执行。

增强功能：
- 当触发 exceeding override 时，区分“零库存跳过”和“目标超出范围”这两类日志消息。

测试：
- 为 `resolveMaxQuantityNext` 添加表驱动测试，覆盖零库存，以及 `maxQuantity` 和 `target` 的边界条件。

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery

Handle zero-inventory scenarios in BetterSliding without treating them as errors and ensure flows terminate cleanly.

Bug Fixes:
- Treat maxQuantity == 0 as an immediate Done state in resolveMaxQuantityNext to prevent tasks from failing or crashing on zero stock.
- Mark zero-stock conditions as exceeded in handleGetMaxQuantity so the flow exits via the override path instead of proceeding incorrectly.

Enhancements:
- Differentiate log messages between zero-inventory skips and out-of-range targets when exceeding override is triggered.

Tests:
- Add table-driven tests for resolveMaxQuantityNext covering zero inventory and boundary conditions for maxQuantity and target.

</details>

### CODE REVIEW

**Comment by zmdyy0318** [comment]
注意，单纯的go测试没有实际意义，需要完成结合pipeline在游戏里跑

File: agent/go-service/bettersliding/handlers.go [hunk 1]
```diff
@@ -233,12 +233,16 @@ func (a *BetterSlidingAction) handleGetMaxQuantity(ctx *maa.Context, arg *maa.Cu
...
-			a.logger.Warn().
+			logEvent := a.logger.Warn().
 				Int("original_target", a.OriginalTarget).
 				Int("resolved_target", a.Target).
 				Int("max_quantity", a.maxQuantity).
-				Str("override_node", a.ExceedingOverrideEnable).
-				Msg("target out of range: exceeding override scheduled, branching to done")
+				Str("override_node", a.ExceedingOverrideEnable)
+			if a.maxQuantity == 0 {
+				logEvent.Msg("stock is zero, skipping sell attempt via exceeding override")
```
Comment (single-ptilopsis) [inline agent/go-service/bettersliding/handlers.go]: 通用组件，msg中不应出现特定任务的描述，如`stock`与`sell`

File: agent/go-service/bettersliding/handlers.go
Comment (single-ptilopsis) [inline agent/go-service/bettersliding/handlers.go:682]: nodeBetterSlidingDone的判断条件可以直接改为 maxQuantity == target，此时已经将数量滑动到可滑动的最大，顺手改下喵🐈

**Comment by ztg-pty** [comment]
对，所以我跑了一遍，好像是没啥问题

**Comment by ztg-pty** [comment]
在测试是我将所有库存物品清空测试在前台模式下是否能解决零库存硬拦截问题，根据日志及游戏内部表现，未发现有相关问题出现
（可能环境不同，结果也可能不同，我的分辨率是1920*1080）

**Review by single-ptilopsis — CHANGES_REQUESTED** [review]

**Comment by zmdyy0318** [comment]
> 对，所以我跑了一遍，好像是没啥问题

没有实际意义的代码建议删掉，不然开了口子，后期代码越来越多

**Comment by ztg-pty** [comment]
> > 没错，所以运行了一遍，似乎没有问题
> 
> 建议删除没有实际意义的代码，避免日后代码越来越多"
好的

**Review by ocsin1 — DISMISSED** [review]
请运行一下命令修改 git 设置,并修改提交信息中的提交者(目前是 MaaEnd Agent)后强制推送,不要拿你用的给 ai 取的名字,不然最后的提交者没有你的账号
还有建议自行学习开发的基本能力,不要太依赖 ai 了
```bash
git config --global user.name "ly-ztg"
git config --global user.email "3783181810@qq.com"
```

**Comment by ztg-pty** [comment]
误推送啊

**Comment by single-ptilopsis** [comment]
改好了的话点一下 Ready for review ❤️

File: agent/go-service/bettersliding/handlers.go
Comment (zmdyy0318) [reply agent/go-service/bettersliding/handlers.go:682]: @copilot 改一下这个

**Comment by zmdyy0318** [comment]
没开Allow edits from maintainers 我没法在上面改

**Comment by zmdyy0318** [comment]
#3231 没法在这个分支改，我新开了pr
